src/opm/parser/eclipse/EclipseState/Schedule/Well/Well2.cpp

```diff
@@ -134,6 +135,17 @@ bool Well2::updateWellGuideRate(double guide_rate_arg) {
 }
 
 
+bool Well2::updateFoamProperties(std::shared_ptr<WellFoamProperties> foam_properties_arg) {
+    if (*this->foam_properties != *foam_properties_arg) {
+        this->foam_properties = foam_properties_arg;
+        this->producer = false;
```

Comment:
This should be changed with a call to the `switchToProducer()` method.

Reply:
I think you are correct in that the WFOAM keyword should not be used for a producer, i.e. this is an error. I'll change the code to check for this. The same applies to the polymer code, I will change that in the same way.

Reply:
Doing this, it turns out that several tests then fail. I think the best thing to do is to modify the tests (typically to include a WCONINJE to make the wells injectors prior to the WPOLYMER or WFOAM).